server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -5455,6 +5456,15 @@ private void encryptAndStorePassword(UserVmVO vm, String password) {
         }
     }
 
+    public void persistDeviceBusInfo(UserVmVO vm, String rootDiskController) {
+        String existingVmRootDiskController = vm.getDetail(VmDetailConstants.ROOT_DISK_CONTROLLER);
+        if (StringUtils.isEmpty(existingVmRootDiskController) && !StringUtils.isEmpty(rootDiskController)) {
+            vm.setDetail(VmDetailConstants.ROOT_DISK_CONTROLLER, rootDiskController);
+            _vmDao.saveDetails(vm);
+            s_logger.debug("Persisted device bus information rootDiskController=" + rootDiskController + " for vm: " + vm.getDisplayName());
```

Comment:
Wrap this ``debug`` call in a ``if (isDebugEnabled()`` block.  Also, please add the vm id to the log message to assist with debugging.